data-collection/deploy/module-agreements.yaml

```diff
@@ -0,0 +1,851 @@
+                  # Delete existing crawler to recreate with new config
+                  try:
+                      glue.get_crawler(Name=crawler_name)
+                      logger.info(f"Deleting existing crawler {crawler_name} to recreate with new config")
+                      glue.delete_crawler(Name=crawler_name)
+                      # Wait a moment for deletion to complete
+                      import time
+                      time.sleep(2)
+                  except glue.exceptions.EntityNotFoundException:
+                      logger.info(f"Crawler {crawler_name} doesn't exist")
+                  except Exception as e:
+                      logger.warning(f"Could not delete crawler {crawler_name}: {e}")
+                  
+                  # Create crawler with new configuration
+                  try:
+                      glue.create_crawler(
+                          Name=crawler_name,
+                          Role=os.environ.get('GLUE_ROLE_ARN', ''),
+                          DatabaseName=os.environ.get('DATABASE_NAME', 'optimization_data'),
+                          TablePrefix=crawler_config['table_prefix'],
+                          Targets={
+                              'S3Targets': [
+                                  {
+                                      'Path': crawler_config['s3_path']
+                                  }
+                              ]
+                          },
+                          SchemaChangePolicy={
+                              'UpdateBehavior': 'UPDATE_IN_DATABASE',
+                              'DeleteBehavior': 'LOG'
+                          },
+                          RecrawlPolicy={
+                              'RecrawlBehavior': 'CRAWL_EVERYTHING'
+                          }
+                      )
+                      logger.info(f"Created crawler {crawler_name} with grouping configuration")
+                  except glue.exceptions.AlreadyExistsException:
+                      logger.info(f"Crawler {crawler_name} already exists, skipping creation")
+                  except Exception as e:
+                      logger.error(f"Failed to create crawler {crawler_name}: {e}")
+                      continue
+                  
+                  # Start the crawler
+                  try:
+                      glue.start_crawler(Name=crawler_name)
+                      logger.info(f"Started crawler {crawler_name}")
+                  except Exception as e:
+                      if "CrawlerRunningException" not in str(e):
+                          logger.warning(f"Could not start crawler {crawler_name}: {e}")
+                      else:
+                          logger.info(f"Crawler {crawler_name} is already running")
+
+          def create_quicksight_datasets():
+              quicksight = boto3.client('quicksight')
+              glue = boto3.client('glue')
+              account_id = boto3.client('sts').get_caller_identity()['Account']
```

Comment:
1. you have account_id in lambda without additional call
2. please do not create datasets here. Some customers can use data collection WITHOUT qs.